compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt

```diff
@@ -279,9 +279,43 @@ internal abstract class ComposeBridge(
         })
     }
 
+    // Decides which AWT events should be delivered, and which should be filtered out
+    private val awtEventFilter = object {
+
+        var isPrimaryButtonPressed = false
+
+        fun shouldSendMouseEvent(event: MouseEvent): Boolean {
+            // AWT can send events after the window is disposed
+            if (isDisposed)
+                return false
+
+            // Filter out mouse events that report a change in the pressed state of the primary
+            // button, but aren't themselves mouse press/release events.
+            // See https://github.com/JetBrains/compose-multiplatform/issues/2850 for details
+            val eventReportsPrimaryButtonPressed =
+                (event.modifiersEx and MouseEvent.BUTTON1_DOWN_MASK) != 0
+            if (eventReportsPrimaryButtonPressed != isPrimaryButtonPressed) {
+                isPrimaryButtonPressed = when (event.id) {
+                    MouseEvent.MOUSE_PRESSED -> true
+                    MouseEvent.MOUSE_RELEASED -> false
+                    else -> return false  // Note the return
+                }
+            }
+
+            return true
+        }
+
+        @Suppress("UNUSED_PARAMETER")
+        fun shouldSendKeyEvent(event: KeyEvent): Boolean {
+            // AWT can send events after the window is disposed
+            return !isDisposed
+        }
+    }
+
     private fun onMouseEvent(event: MouseEvent): Unit = catchExceptions {
```

Comment:
It's called from several places including `mouseEntered`/`mouseExited` events, so it looks like we need to filter it only there and drop additional bool storage

Reply:
We need to know the current pressed state of the primary mouse button in order to filter correctly.